added migrations and upgrade guides docs (#1432)
medusajs/medusa: docs/content/advanced/backend/migrations.md

# Migrations

In this document, you’ll learn about what Migrations are, their purpose, how you can run them, and how you can create your own Migrations.

:::note

Medusa’s Migrations do not work with SQLite databases. They are intended to be used with PostgreSQL databases, which is the recommended Database for your Medusa production server.

:::

## Overview

> Migrations in Medusa are scripts that are used to make additions or changes to your database schema. They are essential for both when you first install your server and for subsequent server upgrades later on.

Reviewer: suggestion: Would make this more general, since migrations in Medusa are no different from migrations in other contexts. I would suggest something like Typeorm's definition: `A migration is just a single file with sql queries to update a database schema and apply new changes to an existing database.`

> When you first install your Medusa server, the database schema used must have all the tables necessary for the server to run. This is automatically done when you run the `seed` command which also seeds your database with dummy data:

Reviewer: comment: Would be great to add both approaches of running migrations.

```suggestion
When you first create your Medusa server, the database schema must have all the tables necessary for the server to run. This can be done in one of two ways: 
- medusa migrations run -> add a couple of sentences...
- medusa seed -> migrations + dummy data...
```

```bash npm2yarn
npm run seed
```

> When a new Medusa version requires changes to your database schema, it will introduce new migration scripts that must run before using the upgraded version. Not running the necessary migrations for a new update will result in a lot of unexpected behaviors and inconsistencies.

Reviewer: nit: Slightly rephrase
```suggestion
When a new Medusa version introduces changes to the database schema, you'll have to run migrations to apply them to your own database.
```

:::tip

> Migrations are used for changes in the database schema. However, there are some version updates of Medusa that require updating the data in your database to fit the new schema or architecture. Those are specific to each version and you should check out the version under Upgrade Guides for details on the steps.

Reviewer: nit: Slightly rephrase
```suggestion
Migrations are used to apply changes to the database schema. However, there are some version updates of Medusa that require updating the data in your database to fit the new schema. Those are specific to each version and you should check out the version under Upgrade Guides for details on the steps.
```
